airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -738,6 +737,23 @@ def list(self, bucket_name, versions=None, max_results=None, prefix=None, delimi
                 break
         return ids
 
+    def list_by_prefix(self, prefix: str | List[str] | None, bucket_name: str, delimiter: str | None = None):
+        """
+        List all objects from the bucket with the given prefix or list of prefixes
+
+        :param bucket_name: bucket name
+        :param prefix: string or list of strings which filter objects whose name begin with it/them
+        :param delimiter: filters objects based on the delimiter (for e.g '.csv')
+        :return: a stream of object names matching the filtering criteria
+        """
+        objects = []
+        if isinstance(prefix, list):
+            for prefix_item in prefix:
+                objects.extend(self.list(bucket_name=bucket_name, prefix=prefix_item, delimiter=delimiter))
+        else:
+            objects.extend(self.list(bucket_name=bucket_name, prefix=prefix, delimiter=delimiter))
+        return objects
```

Comment:
Do we need a new function for it?
Can't we add `List[str]` support in `prefix()` function?

Reply:
Good catch, it indeed makes more sense to encapsulate the existing `list` method within an inner function and convert the `list_by_prefix` into the `list` method. Working on it.